lib/fs.js

```diff
@@ -2014,30 +2014,40 @@ function ReadStream(path, options) {
   this.closed = false;
 
   if (this.start !== undefined) {
-    if (typeof this.start !== 'number' || Number.isNaN(this.start)) {
-      throw new ERR_INVALID_ARG_TYPE('start', 'number', this.start);
-    }
-    if (this.end === undefined) {
-      this.end = Infinity;
-    } else if (typeof this.end !== 'number' || Number.isNaN(this.end)) {
-      throw new ERR_INVALID_ARG_TYPE('end', 'number', this.end);
+    if (!Number.isSafeInteger(this.start)) {
+      if (typeof this.start !== 'number')
+        throw new ERR_INVALID_ARG_TYPE('start', 'number', this.start);
+      if (!Number.isInteger(this.start))
+        throw new ERR_OUT_OF_RANGE('start', 'an integer', this.start);
+      throw new ERR_OUT_OF_RANGE('start', '=> 0 and <= 2 ** 53 - 1', this.start);
     }
-
-    if (this.start > this.end) {
-      const errVal = `{start: ${this.start}, end: ${this.end}}`;
-      throw new ERR_OUT_OF_RANGE('start', '<= "end"', errVal);
+    if (this.start < 0) {
+      throw new ERR_OUT_OF_RANGE('start', '=> 0 and <= 2 ** 53 - 1', this.start);
```

Comment:
You were right about changing => to >= below but that should also be done here :-)

Reply:
Good catch! I'm sorry to have missed that out, changing ASAP.

Reply:
Ligatures to the rescue!
<img width="898" alt="screen shot 2018-05-09 at 1 22 58 am" src="https://user-images.githubusercontent.com/13121330/39779680-905bbca2-5327-11e8-9509-d609205a4960.png">